packages/jest-worker/src/workers/NodeThreadsWorker.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+'use strict';
+
+import {
+  CHILD_MESSAGE_INITIALIZE,
+  PARENT_MESSAGE_OK,
+  PARENT_MESSAGE_CLIENT_ERROR,
+  PARENT_MESSAGE_SETUP_ERROR,
+} from '../types';
+
+import type {Readable} from 'stream';
+import type {
+  ChildMessage,
+  OnEnd,
+  OnStart,
+  WorkerOptions,
+  WorkerInterface,
+} from '../types';
+
+// $FlowFixMe: Flow doesn't know about experimental features of Node
+const {Worker} = require('worker_threads');
+
+export default class ExperimentalWorker implements WorkerInterface {
+  _worker: Worker;
+  _options: WorkerOptions;
+  _onProcessEnd: OnEnd;
+  _retries: number;
+
+  constructor(options: WorkerOptions) {
+    this._options = options;
+    this.initialize();
+  }
+
+  initialize() {
+    this._worker = new Worker(__dirname + '/threadChild.js', {
+      eval: false,
+      stderr: true,
+      stdout: true,
+      // $FlowFixMe: Flow does not work well with Object.assign.
+      workerData: Object.assign(
+        {
+          cwd: process.cwd(),
+          env: Object.assign({}, process.env, {
+            JEST_WORKER_ID: this._options.workerId,
+          }),
+          // Suppress --debug / --inspect flags while preserving others (like --harmony).
+          execArgv: process.execArgv.filter(v => !/^--(debug|inspect)/.test(v)),
+          silent: true,
+        },
+        this._options.forkOptions,
+      ),
+    });
+
+    this._worker.on('message', this.onMessage.bind(this));
+    this._worker.on('exit', this.onExit.bind(this));
+
+    this._worker.postMessage([
+      CHILD_MESSAGE_INITIALIZE,
+      false,
+      this._options.workerPath,
+      this._options.setupArgs,
+    ]);
+
+    this._retries++;
+
+    // If we exceeded the amount of retries, we will emulate an error reply
+    // coming from the child. This avoids code duplication related with cleaning
+    // the queue, and scheduling the next call.
+    if (this._retries > this._options.maxRetries) {
+      const error = new Error('Call retries were exceeded');
+
+      this.onMessage([
+        PARENT_MESSAGE_CLIENT_ERROR,
+        error.name,
+        error.message,
+        error.stack,
+        {type: 'WorkerError'},
+      ]);
+    }
+  }
+
+  onMessage(response: any /* Should be ParentMessage */) {
```

Comment:
Add the real type, but a `$FlowFixMe` above instead?

Reply:
Doesn't work, I changed back to any because having flowfixme every other line is worse